### PR TITLE
jetty 10

### PR DESCRIPTION
**Pull request by dev-mlb**
or #550

### CODE REVIEW

File: src/main/java/emissary/server/EmissaryServer.java
Comment (fbruton) [inline src/main/java/emissary/server/EmissaryServer.java:1]: Remove throws `GeneralSecurityException` from  the `configureServer()` method.  Code was updated to catch and throw an `IOException`.

**Review by fbruton — CHANGES_REQUESTED** [review]
This transitive dependency is  a bit concerning. Does jersey need to be updated to 3.x?
```
[INFO] +- org.glassfish.jersey.containers:jersey-container-jetty-http:jar:2.40:compile
[INFO] |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
[INFO] |  +- org.eclipse.jetty:jetty-continuation:jar:9.4.51.v20230217:compile
```

File: src/main/java/emissary/server/EmissaryServer.java
Comment (dev-mlb) [reply src/main/java/emissary/server/EmissaryServer.java:1]: Done

**Comment by dev-mlb** [comment]
> This transitive dependency is a bit concerning. Does jersey need to be updated to 3.x?
> 
> ```
> [INFO] +- org.glassfish.jersey.containers:jersey-container-jetty-http:jar:2.40:compile
> [INFO] |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
> [INFO] |  +- org.eclipse.jetty:jetty-continuation:jar:9.4.51.v20230217:compile
> ```

Yeah, it's unfortunate but this code is deprecated and not in the next version of Jetty due to the new Async stuff. It is just this jar, which has only a few classes in it. If I upgrade to Jersey 3+, the Jakarta conversion starts which would probably be easier to take #550...

**Comment by jpdahlke** [comment]
> > This transitive dependency is a bit concerning. Does jersey need to be updated to 3.x?
> > ```
> > [INFO] +- org.glassfish.jersey.containers:jersey-container-jetty-http:jar:2.40:compile
> > [INFO] |  +- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
> > [INFO] |  +- org.eclipse.jetty:jetty-continuation:jar:9.4.51.v20230217:compile
> > ```
> 
> Yeah, it's unfortunate but this code is deprecated and not in the next version of Jetty due to the new Async stuff. It is just this jar, which has only a few classes in it. If I upgrade to Jersey 3+, the Jakarta conversion starts which would probably be easier to take #550...

I think I would like to try going forward with Jetty 10 first even with this jar.

**Review by cfkoehler — APPROVED** [review]
I was not sure the best way to review this.
I started up a server with and without the new SniHost option and all seems to work as expected.

Also started up multiple emissary as a local cluster and that all seemed correct.

All the pom changes make sense when looking at the dependency tree.

**Review by jpdahlke — APPROVED** [review]
I integrated this downstream and don't see any immediate issues.

**Pull request merged**